pkg/plugins/driver.proto

```diff
@@ -0,0 +1,176 @@
+syntax = "proto3";
+
+package plugins;
+
+option go_package = "github.com/lima-vm/lima/pkg/plugins";
+
+service VMDriver {
+    rpc GetMetadata(GetMetadataRequest) returns (GetMetadataResponse);
+    rpc StartVM(StartVMRequest) returns (StartVMResponse);
+    rpc StopVM(StopVMRequest) returns (StopVMResponse);
+    rpc Initialize(InitializeRequest) returns (InitializeResponse);
+    rpc CreateDisk(CreateDiskRequest) returns (CreateDiskResponse);
+    rpc Validate(ValidateRequest) returns (ValidateResponse);
+    rpc Register(RegisterRequest) returns (RegisterResponse);
+    rpc Unregister(UnregisterRequest) returns (UnregisterResponse);
+    rpc ChangeDisplayPassword(ChangeDisplayPasswordRequest) returns (ChangeDisplayPasswordResponse);
+    rpc GetDisplayConnection(GetDisplayConnectionRequest) returns (GetDisplayConnectionResponse);
+    rpc CreateSnapshot(CreateSnapshotRequest) returns (CreateSnapshotResponse);
+    rpc ApplySnapshot(ApplySnapshotRequest) returns (ApplySnapshotResponse);
+    rpc DeleteSnapshot(DeleteSnapshotRequest) returns (DeleteSnapshotResponse);
+    rpc ListSnapshots(ListSnapshotsRequest) returns (ListSnapshotsResponse);
+    rpc GetGuestAgentConnection(GetGuestAgentConnectionRequest) returns (GetGuestAgentConnectionResponse);
+}
+
+// Response is a common response type used by helper functions
+message Response {
+    bool success = 1;
+    string message = 2;
+}
+
+message GetMetadataRequest {}
+
+message GetMetadataResponse {
+    string name = 1;
+    string version = 2;
+    string description = 3;
+    repeated string supported_vm_types = 4;
```

Comment:
This is interesting topic.

- Is there any use case to support multiple VM types in a single plugin? (e.g., "qemu" plugin may support both "qemu" and "qemu-utm" VM types?)
- Also, is there any use case to support the same VM type in multiple plugins?